roles/edpm_deploy/tasks/main.yml

```diff
@@ -136,34 +136,35 @@
       ansible.builtin.debug:
         var: cifmw_edpm_deploy_crs_kustomize_result
 
-    - name: Apply the OpenStackDataPlaneNodeSet CR
-      when: not cifmw_edpm_deploy_dryrun
-      cifmw.general.ci_script:
-        output_dir: "{{ cifmw_edpm_deploy_basedir }}/artifacts"
-        script: "oc apply -f {{ cifmw_edpm_deploy_crs_kustomize_result.output_path }}"
+    - name: Apply dataplane resources but ignore DataPlaneDeployment
+      vars:
+        _timeout_min_string: "{{cifmw_edpm_deploy_timeout + 'm'}}"
+      kubernetes.core.k8s:
+        kubeconfig: "{{ cifmw_openshift_kubeconfig  }}"
+        api_key: "{{ cifmw_openshift_token | default(omit) }}"
+        context: "{{ cifmw_openshift_context | default(omit)  }}"
+        state: present
+        definition: "{{ lookup('file', cifmw_edpm_deploy_crs_kustomize_result.output_path) | from_yaml_all | rejectattr('kind', 'search', cifmw_edpm_deploy_step2_kind }}"
+        wait: true
+        wait_condition:
```

Comment:
not sure if is goint to work, since we are creating multiple resources here, not all have this condition type.
the other approach is to use another task with k8s_info + wait_condition.